### ISSUE TITLE
[Select] Discrepancy in component behavior

### ISSUE THREAD

**Issue by Kritvi-bhatia17** [carbon-design-system/carbon#16776]
### Package

carbon-components

### Package version

1.59.0

### React version

18.2.0

### Description

The Select component should behave as shown in the 1st image (Figma kit) when any other option is selected. However, in [Storybook](https://react.carbondesignsystem.com/?path=/story/components-select--playground), it is currently displaying as shown in the 2nd image.

![image](https://github.com/carbon-design-system/carbon/assets/158570656/99d7208d-3204-4799-bf94-242c93bd1c5c)

![image](https://github.com/carbon-design-system/carbon/assets/158570656/cb1db461-6c22-424e-9b27-c06f9db9b280)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-select--playground

### Steps to reproduce

- [ ] Update the update the example text for the options accordingly.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
The option with the lengthy text string is so that we can validate the truncation ellipsis works as expected

![Image](https://github.com/carbon-design-system/carbon/assets/3360588/b08686b2-42fc-4bdc-b5e1-85d7da6b2e18)

We could change this to be lower in the option list, but arguably this long string example should be present in figma to showcase the truncation ellipsis functionality to match storybook.

**Comment by Kritvi-bhatia17** [comment]
Yes, the long text is fine, and it's a smart idea to show the same in the Figma kit as well. 

However, regarding the issue I mentioned: when the first option is deselected, the text should appear. 
Let me confirm internally with the team if we want to simplify the interaction by always displaying the placeholder value.

Uploading Screen Recording 2024-06-13 at 9.21.05 PM.mov…

**Comment by Kritvi-bhatia17** [comment]
I realized the Select component's placeholder text has already been discussed with Michael (in this [issue](https://github.com/carbon-design-system/carbon/issues/13008). He had specific reasons to remove the default "Choose an option" and add an empty default option.

For now, we can match the interaction with the Figma component. `Initially, it should show a blank state, and when any other option is selected, the first option should become visible.`
